mobile/src/main/java/org/openhab/habdroid/ui/activity/ContentController.java

```diff
@@ -412,6 +415,18 @@ public void onWidgetUpdated(OpenHABWidget widget) {
         }
     }
 
+    @Override
+    public void onLoadFailure(int statusCode, Headers headers, byte[] responseBody, Throwable e) {
+        int errorMsgRes = R.string.error_sitemap_generic_load_error;
+        if (statusCode == HttpURLConnection.HTTP_NOT_FOUND) {
+            errorMsgRes = R.string.error_sitemap_not_found;
+        }
```

Comment:
Ternary operator would seem more appropriate here. In any case, please add StringRes annotation.

Also, wouldn't it make sense to allow retry instead of opening drawer in case of non-404 failure?

Reply:
+1

I thought about the retry option, too, but I wasn't sure, if it's a valid option all cases. But if you think, a retry for non-404 cases is ok, too, I'm with you, that this is the better option :)

Reply:
It probably won't help in most cases (e.g. 401 or 403), but still (IMHO) is a more useful action than opening the drawer.